app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt

```diff
@@ -37,15 +41,21 @@ private const val BUNDLE_CHAT_ROOM_ID = "chat_room_id"
 private const val BUNDLE_CHAT_ROOM_NAME = "chat_room_name"
 private const val BUNDLE_CHAT_ROOM_TYPE = "chat_room_type"
 private const val BUNDLE_IS_CHAT_ROOM_READ_ONLY = "is_chat_room_read_only"
+private const val REQUEST_CODE_FOR_PERFORM_SAF = 42
 
 class ChatRoomFragment : Fragment(), ChatRoomView {
-
     @Inject lateinit var presenter: ChatRoomPresenter
     private lateinit var chatRoomId: String
     private lateinit var chatRoomName: String
     private lateinit var chatRoomType: String
     private var isChatRoomReadOnly: Boolean = false
     private lateinit var adapter: ChatRoomAdapter
+    // For reveal and unreveal anim.
+    private val hypotenuse by lazy { Math.hypot(relative_layout.width.toDouble(), relative_layout.height.toDouble()).toFloat() }
+    private val max by lazy { Math.max(layout_message_attachment_options.width.toDouble(), layout_message_attachment_options.height.toDouble()).toFloat() }
+    private val centerX by lazy { recycler_view.right }
+    private val centerY by lazy { recycler_view.bottom }
+    private lateinit var handler: Handler
```

Comment:
You can initialize the handler here. As a immutable `val`

Reply:
Why not after the fragment (on `onViewCreated()`) is created?

Reply:
Because it doesn't need to wait until `onViewCreated` and it can be immutable...

Reply:
Because we are using the handler after `onViewCreated()` is called we don't need to keep that variable member in the memory when the `Fragment` is created.

Maybe initialising the handler as proposed by you has a minimum impact, but I believe that we don't need to create it if we are using only after the UI is created...

Reply:
It won't have any noticeable impact, one because it will be created when we need the fragment, doesn't matter if on `onViewCreated` or on the Object creation, and because it is a simple object instantiation...

Ideally we should use immutables and readonly `val`s whenever it is possible, since it prevents all sort of bugs down the road.

>Classes should be immutable unless there’s a very good reason to make them mutable.
Joshua Block, Effective Java